spinnaker_graph_front_end/examples/hello_world_untimed/hello_world_vertex.py

```diff
@@ -0,0 +1,133 @@
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from enum import Enum
+import logging
+from spinn_utilities.overrides import overrides
+from pacman.model.graphs.machine import MachineVertex
+from pacman.model.resources import ResourceContainer, VariableSDRAM
+from spinn_front_end_common.utilities.constants import (
+    SYSTEM_BYTES_REQUIREMENT, BYTES_PER_WORD)
+from spinn_front_end_common.utilities.helpful_functions import (
+    locate_memory_region_for_placement)
+from spinn_front_end_common.abstract_models import (
+    AbstractGeneratesDataSpecification)
+from spinn_front_end_common.interface.buffer_management.buffer_models import (
+    AbstractReceiveBuffersToHost)
+from spinn_front_end_common.interface.buffer_management import (
+    recording_utilities)
+from spinnaker_graph_front_end.utilities import SimulatorVertex
+from spinnaker_graph_front_end.utilities.data_utils import (
+    generate_steps_system_data_region)
+import numpy
+from pacman.executor.injection_decorator import inject_items
+
+logger = logging.getLogger(__name__)
+
+
+class HelloWorldVertex(
+        SimulatorVertex, AbstractGeneratesDataSpecification,
+        AbstractReceiveBuffersToHost):
+
+    DATA_REGIONS = Enum(
+        value="DATA_REGIONS",
+        names=[('SYSTEM', 0),
+               ('PARAMS', 1),
+               ('STRING_DATA', 2)])
+
+    PARAMS_BASE_SIZE = BYTES_PER_WORD * 2
+
+    def __init__(self, label, constraints=None):
+        """
+
+        :param int n_repeats: The number of times to repeat the label in total
+        :param str label: The label, which will be printed
+        :param constraints: The constraints of the vertex
+        """
+        super(HelloWorldVertex, self).__init__(
+            label, "hello_world.aplx", constraints=constraints)
+
+        # Make the text fit at a word boundary
+        self._text = label
+        text_extra = len(label) % BYTES_PER_WORD
+        if text_extra != 0:
+            for _ in range(BYTES_PER_WORD - text_extra):
+                self._text += ' '
+
+    @property
+    @overrides(MachineVertex.resources_required)
+    def resources_required(self):
+        fixed = (
+            SYSTEM_BYTES_REQUIREMENT +
+            recording_utilities.get_recording_header_size(1) +
+            self.PARAMS_BASE_SIZE + len(self._text))
+        variable = len(self._text)
```

Comment:
i do admit im hitting a issue that this reocrding isnt designed for what hello world really wants...... not sure how to fix that...... unless you augment variable sdram to know its only used it x times. or something

Reply:
Is isn't wrong any more, because I changed it to not stop.  It now prints the text every step, so if you run for 10 steps, it prints it 10 times.  It doesn't stop arbitrarily any more.

Reply:
k then